src/components/recommendations/recommendations.js

```diff
@@ -0,0 +1,47 @@
+import './recommendations.scss';
+import { RECO_FBT_TEMPLATE } from './templates';
+import handlebars from 'handlebars';
+import { observeStoreByKey } from '../../store';
+import { validateContainer } from '../../util/dom';
+import { clearRecommendation } from "../../actions/recommendations";
+
+export const TYPE_FREQUENTLY_BOUGHT_TOGETHER = 'FREQUENTLY_BOUGHT_TOGETHER';
+export default class Recommendations {
+
+  constructor(client, reduxStore, conf, recommendationSettings) {
+    this.client = client;
+    this.conf = conf;
+    this.reduxStore = reduxStore;
+    recommendationSettings.push(conf);
+
+    handlebars.registerHelper('test', (item) => {
+      window.console.log('+++ item', item);
+      return item;
+    });
+
+    if (validateContainer(conf.containerId)) {
+      observeStoreByKey(this.reduxStore, 'recommendation', (res) => {
+        if (res.container === conf.containerId) {
+          this.render(res);
```

Comment:
shouldn't `render()` be invoked automatically when the state changes? maybe I'm missing something here.

Reply:
That's a design principle for React. In Search UI Library, we define a render method for each component, then observe relating state changes to call the render method.